src/main/java/teammates/ui/controller/InstructorSearchPageAction.java

```diff
@@ -110,22 +105,30 @@ protected ActionResult execute() throws EntityDoesNotExistException {
     private int filterFeedbackResponseCommentResults(
             FeedbackResponseCommentSearchResultBundle frCommentSearchResults,
             List<InstructorAttributes> instructors, int totalResultsSize) {
+        
         Iterator<Entry<String, List<FeedbackResponseAttributes>>> iterFr = frCommentSearchResults.responses.entrySet().iterator();
+        
         while (iterFr.hasNext()) {
             List<FeedbackResponseAttributes> frs = iterFr.next().getValue();
             Iterator<FeedbackResponseAttributes> fr = frs.iterator();
+            
             while (fr.hasNext()) {
                 FeedbackResponseAttributes response = fr.next();
                 FeedbackQuestionAttributes relatedQuestion = getRelatedQuestionOfResponse(frCommentSearchResults, response);
                 InstructorAttributes instructor = this.getInstructorForCourseId(response.courseId, instructors);
+                
                 boolean isVisibleResponse = true;
-                boolean needCheckPrivilege = relatedQuestion == null || !(relatedQuestion.recipientType == FeedbackParticipantType.NONE ||
-                        relatedQuestion.recipientType == FeedbackParticipantType.INSTRUCTORS ||
-                                relatedQuestion.recipientType == FeedbackParticipantType.STUDENTS);
-                boolean isNotAllowedForInstructor = instructor == null || !(instructor.isAllowedForPrivilege(response.giverSection,
-                        response.feedbackSessionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS))
-                        || !(instructor.isAllowedForPrivilege(response.recipientSection,
-                                response.feedbackSessionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS));
+                boolean needCheckPrivilege = relatedQuestion == null 
+                                              || !(relatedQuestion.recipientType == FeedbackParticipantType.NONE 
+                                                   || relatedQuestion.recipientType == FeedbackParticipantType.INSTRUCTORS 
+                                                   || relatedQuestion.recipientType == FeedbackParticipantType.STUDENTS);
+                
```

Comment:
Can you help me remove this variable (needCheckPrivilege). I missed this out in #2982.